cmd/kube-controller-manager/app/core.go

```diff
@@ -105,6 +105,12 @@ func startNodeIpamController(ctx context.Context, controllerContext ControllerCo
 		return nil, false, nil
 	}
 
+	// Cannot run cloud ipam controller if cloud provider is nil (not set or 'external')
+	if controllerContext.Cloud == nil && controllerContext.ComponentConfig.KubeCloudShared.CIDRAllocatorType == string(ipam.CloudAllocatorType) {
+		klog.Warningf("--cidr-allocator-type is set to 'CloudAllocator' but cloud provider is not configured")
```

Comment:
I see that in https://github.com/kubernetes/kubernetes/issues/114556#issuecomment-1358736436 you are mentioning that we can consider quitting KCM in case of misconfiguratuion. I think this is a great idea.

That would be a clear signal that KCM is misconfigured, with an explanation what configuration should be used. It seems a better solution than providing half working cluster that will generate operational load when deployed.

Reply:
agreed -- I updated to error instead of just warning